src/php/inc/theme-setup.php

```diff
@@ -95,3 +95,50 @@ function custom_hide_editor() {
 	}
 }
 add_action( 'admin_init', 'custom_hide_editor' );
+
+/**
+ * Customize the pingbacks and trackbacks option on new posts by default.
+ *
+ * @param mixed  $value   The current option value.
+ * @param string $option The name of the option being filtered.
+ * @param mixed  $default The default value for the option.
+ *
+ * @return mixed The filtered option value. If the provided option is 'default_ping_status',
+ *               this function will set its value to 'closed' (disabling pingbacks and trackbacks),
+ *               and return the updated value. For other options, it returns the original value.
+ */
+function custom_disable_pingbacks_trackbacks_option( $value, $option, $default ) {
+    if ( 'default_ping_status' === $option ) {
+        $value = 'closed';
+    }
+
+    return $value;
+}
+
+add_filter( 'pre_option_default_ping_status', 'custom_disable_pingbacks_trackbacks_option', 10, 3 );
+
+/**
+ * Disable pingbacks and trackbacks on the default Hello world post.
+ */
+function update_hello_world_post_on_init() {
+	$hello_world_post = get_post( 1 );
+
+	if ( $hello_world_post ) {
+		$updated_post_data = array(
+            'ID' => $hello_world_post->ID,
+            'ping_status' => 'closed',
+		);
+
+		$post_id = wp_update_post( $updated_post_data );
+	}
+}
+add_action( 'init', 'update_hello_world_post_on_init' );
+
+/**
+ * Remove Trackbacks Support for the "post" Post Type.
+ */
+function remove_comments_support() {
+	remove_post_type_support( 'post', 'trackbacks' );
```

Comment:
You would think that if you can remove this checkbox from the post, that you could remove checkboxes from the Discussion Settings as well, but I didn't find a way.

Reply:
I found a snippet that seems to do the trick, but it does like 10 different things, so it might be a little heavy-handed. We could include it and maybe make it depend on an environment variable, like only run it if `COMMENTS_ENABLED=false`. Another option would be to document how to turn off comments for all future posts through the admin, and include that in basic setup instructions, kind of like we describe how to change the permalinks. Anyway, here's that snippet:

```php
add_action('admin_init', function () {
    // Redirect any user trying to access comments page
    global $pagenow;

    if ($pagenow === 'edit-comments.php') {
        wp_safe_redirect(admin_url());
        exit;
    }

    // Remove comments metabox from dashboard
    remove_meta_box('dashboard_recent_comments', 'dashboard', 'normal');

    // Disable support for comments and trackbacks in post types
    foreach (get_post_types() as $post_type) {
        if (post_type_supports($post_type, 'comments')) {
            remove_post_type_support($post_type, 'comments');
            remove_post_type_support($post_type, 'trackbacks');
        }
    }
});

// Close comments on the front-end
add_filter('comments_open', '__return_false', 20, 2);
add_filter('pings_open', '__return_false', 20, 2);

// Hide existing comments
add_filter('comments_array', '__return_empty_array', 10, 2);

// Remove comments page in menu
add_action('admin_menu', function () {
    remove_menu_page('edit-comments.php');
});

// Remove comments links from admin bar
add_action('init', function () {
    if (is_admin_bar_showing()) {
        remove_action('admin_bar_menu', 'wp_admin_bar_comments_menu', 60);
    }
});
```